### PR TITLE
Move to PackageES SDL Pipeline template for security and compliance

### PR DESCRIPTION
**Pull request by kmahone**
The PkgESSecComp@10 task that we were previously using is going away in favor of a set of Pipeline templates.

We update our Release Build Pipeline to call these templates. There are two templates:

Windows.SDL.Sources.Analysis.OS.Undocked.yml
This does source level analysis of the repo. It runs tools such as CredScan, Policheck and anti-malware tools. It does not depend on the built binaries, so it runs in parallel to the build in its own Job.

Windows.SDL.Binary.Analysis.OS.Undocked.yml
This does scanning of the binaries produced by the build. It runs at the end of the build job. It runs tools such as Component Governance, BinSkim, etc.

These yml templates will get updated by PackageES team as needed and will add more tools to them as times goes by. When changes get made to these templates our builds will pick them up automatically, so we are always up to date.

### CODE REVIEW

File: build/MUX-Release.yml [hunk 1]
```diff
@@ -2,25 +2,49 @@ name: $(BuildDefinitionName)_$(date:yyMM).$(date:dd)$(rev:rrr)
...
+  steps:
+
+  - checkout: self
+
+  - task: NodeTool@0
+
+  - template: v2/Steps/PackageES/Windows.SDL.Sources.Analysis.OS.Undocked.yml@templates_onebranch
+    parameters:
+      globalsdl:
+        codeql:                                          # run codeql
```
Comment (llongley) [inline build/MUX-Release.yml:30]: The comment says "run codeql" but you've got "enable: false".  Is that intended?

File: .config/tsaoptions.json [hunk 2]
```diff
@@ -0,0 +1,9 @@
+{
```
Comment (llongley) [inline .config/tsaoptions.json:1]: What's this file?

File: test/MUXExperimentalTest/MUXExperimentalTestApp/MUXExperimentalTestApp.csproj [hunk 3]
```diff
@@ -48,7 +48,7 @@
     <UseVSHostingProcess>false</UseVSHostingProcess>
     <ErrorReport>prompt</ErrorReport>
     <Prefer32Bit>true</Prefer32Bit>
-    <UseDotNetNativeToolchain>true</UseDotNetNativeToolchain>
+    <UseDotNetNativeToolchain>false</UseDotNetNativeToolchain>
```
Comment (llongley) [inline test/MUXExperimentalTest/MUXExperimentalTestApp/MUXExperimentalTestApp.csproj:51]: Why these changes?

File: test/TestAppCX/TestAppCX.vcxproj [hunk 4]
```diff
@@ -124,6 +124,11 @@
     <Import Project="$(UserRootDir)\Microsoft.Cpp.$(Platform).user.props" Condition="exists('$(UserRootDir)\Microsoft.Cpp.$(Platform).user.props')" Label="LocalAppDataPlatform" />
   </ImportGroup>
   <PropertyGroup Label="UserMacros" />
+  <ItemDefinitionGroup>
+    <ClCompile>
+      <ControlFlowGuard Condition="'$(Configuration)'=='Release'">Guard</ControlFlowGuard>
```
Comment (llongley) [inline test/TestAppCX/TestAppCX.vcxproj:129]: What does this do?

**Review by llongley — CHANGES_REQUESTED** [review]

File: test/TestAppCX/TestAppCX.vcxproj (hunk 4 above)
Comment (kmahone) [reply test/TestAppCX/TestAppCX.vcxproj:129]: This enables ControlFlowGuard which is a security feature. Every time a virtual call is made, the address of the function is validated against a known good list of functions. It prevents jumping to an invalid location. This is one of the things that BinSkim checks for. Since this is only a test app, this flag is less important. But due to how the template works, there is no easy way to scan only certain files and ignore others. So it is easier to just enable this flag in the test apps also.

File: test/MUXExperimentalTest/MUXExperimentalTestApp/MUXExperimentalTestApp.csproj (hunk 3 above)
Comment (kmahone) [reply test/MUXExperimentalTest/MUXExperimentalTestApp/MUXExperimentalTestApp.csproj:51]: For some reason, BinSkim was choking on this test app. It looks like it was not able to handle apps built with this enabled. Since there is no way in the template to exclude this exe from scanning, this was the easiest way to resolve the issue.

File: .config/tsaoptions.json (hunk 2 above)
Comment (kmahone) [reply .config/tsaoptions.json:1]: It is used by the sdl template. More info here:
https://www.osgwiki.com/wiki/Package_ES_SDL_Templates#TSA_config

Comment (kmahone) [reply .config/tsaoptions.json:1]: Specifically, it controls the bugs that get filed when an issue is found. For example, if PoliCheck hits an issue, the Pipeline will file a bug in our path for us to triage.

File: build/MUX-Release.yml (hunk 1 above)
Comment (kmahone) [reply build/MUX-Release.yml]: Looks like it supports Javascript, TypeScript, Python and Powershell. We only use powershell, so I've enabled that.

**Review by llongley — APPROVED** [review]
